### ISSUE TITLE
Style plot background not working properly

### ISSUE THREAD

**Issue by GdMacmillan** [pyviz/holoviews#3610]
I am using holoviews with matplotlib and plotly backends.
```
import holoviews as hv
import numpy as np

from functools import reduce
hv.extension('plotly', 'matplotlib')
```
The problem is I specify bgcolor in the options of the final overlay. The final overlay is composed of many objects

The code i use to create the plot object looks like the following:
```
plot_obj = reduce(lambda x, y: x * y,
            [
                group.opts(hv.opts.Path3D(color=hv.Cycle(color_swatch)),
                           hv.opts.Scatter3D(color=hv.Cycle(color_swatch), size=10, marker='cross'))
                for group in groups
            ] + 
            [
                arrow.opts(hv.opts.Path3D(color='black'))
            ] +
            [
                prism_points.opts(hv.opts.Scatter3D(color=hv.Cycle(ms_color_swatch), size=4))
            ] +
            [
                unit_vectors.opts(hv.opts.Path3D(color='black'))
            ]
      )
plot_obj.opts(width=900, height=600, aspect=aspect, bgcolor='lightgray')
```

The result is a plot with a white background. Something is weird.
`print(plot_obj)`
:Overlay
   .Conductors.P1_c1_m0_m1 :Path3D   [x,y,z]
   .Conductors.P2_c1_m0_m1 :Path3D   [x,y,z]
   .Conductors.P3_c1_m0_m1 :Path3D   [x,y,z]
   .Conductors.P4_c2_m0_m1 :Path3D   [x,y,z]
   .Conductors.P5_c2_m0_m1 :Path3D   [x,y,z]
   .Conductors.P6_c2_m0_m1 :Path3D   [x,y,z]
   .Span_Points.I          :Scatter3D   [x,y,z]
   .Span_Points.II         :Scatter3D   [x,y,z]
   .Span_Points.III        :Scatter3D   [x,y,z]
   .Span_Points.IV         :Scatter3D   [x,y,z]
   .Span_Points.V          :Scatter3D   [x,y,z]
   .Span_Points.VI         :Scatter3D   [x,y,z]
   .Conductors.P1_c1_m1_m2 :Path3D   [x,y,z]
   .Conductors.P2_c1_m1_m2 :Path3D   [x,y,z]
   .Conductors.P3_c1_m1_m2 :Path3D   [x,y,z]
   .Conductors.P4_c2_m1_m2 :Path3D   [x,y,z]
   .Conductors.P5_c2_m1_m2 :Path3D   [x,y,z]
   .Conductors.P6_c2_m1_m2 :Path3D   [x,y,z]
   .Span_Points.VII        :Scatter3D   [x,y,z]
   .Span_Points.VIII       :Scatter3D   [x,y,z]
   .Span_Points.IX         :Scatter3D   [x,y,z]
   .Span_Points.X          :Scatter3D   [x,y,z]
   .Span_Points.XI         :Scatter3D   [x,y,z]
   .Span_Points.XII        :Scatter3D   [x,y,z]
   .Conductors.P1_c1_m2_m3 :Path3D   [x,y,z]
   .Conductors.P2_c1_m2_m3 :Path3D   [x,y,z]
   .Conductors.P3_c1_m2_m3 :Path3D   [x,y,z]
   .Conductors.P4_c2_m2_m3 :Path3D   [x,y,z]
   .Conductors.P5_c2_m2_m3 :Path3D   [x,y,z]
   .Conductors.P6_c2_m2_m3 :Path3D   [x,y,z]
   .Span_Points.XIII       :Scatter3D   [x,y,z]
   .Span_Points.XIV        :Scatter3D   [x,y,z]
   .Span_Points.XV         :Scatter3D   [x,y,z]
   .Span_Points.XVI        :Scatter3D   [x,y,z]
   .Span_Points.XVII       :Scatter3D   [x,y,z]
   .Span_Points.XVIII      :Scatter3D   [x,y,z]
   .Total_Station.I        :Path3D   [x,y,z]
   .Device_Points.Ms1      :Scatter3D   [x,y,z]
   .Device_Points.Ms2      :Scatter3D   [x,y,z]
   .Unit_Vectors.I         :Path3D   [x,y,z]
   .Unit_Vectors.II        :Path3D   [x,y,z]

**Comment by jbednar** [comment]
I can't tell what behavior is desired here; is the 'bgcolor' option meant to apply to the matplotlib or to the plotly backend, and which one are you actually using?  In general, options are backend specific.

**Comment by GdMacmillan** [comment]
Ah that makes sense, it appears i can remove the matplotlib extension and just use plotly. The bgcolor option appears to be accessible but it isn't working.

**Comment by jbednar** [comment]
Yes, usually it's best just to use a single backend at a time, unless you have unusual needs, which helps improve validation (as you'll get warned when you use an option for the wrong backend).

**Comment by GdMacmillan** [comment]
Oh sorry i didn't get it to work. i just removed the matplotlib backend and am using only plotly. I still need help with this if possible.

**Comment by jbednar** [comment]
What's the problem exactly, then?  Did you want the bgcolor option only to be applied when the object is used with matplotlib?  If so, pull all your matplotlib-specific options into one .opts() call and add `backend='matplotlib'` to that call, and have the rest of them be plotly options (the default backend you selected above).  Then the mpl options should be applied when using mpl, and the plotly options when using plotly.

**Comment by GdMacmillan** [comment]
The problem is specifying bgcolor is not changing the background color in the plot to light grey even though it is shown as light grey in the layout variable. The plot shows up with just a white background.

```
fig = hv.render(plot_obj)
fig.get('layout')
```

{'width': 900,
 'height': 600,
 'title': '',
 'plot_bgcolor': 'lightgrey',
 'scene': {'xaxis': {'range': [-518.0344, 427.0426], 'title': 'x'},
  'yaxis': {'range': [-102.6931, 127.73680000000002], 'title': 'y'},
  'zaxis': {'range': [-5.981263867770538, 50.72950000000003], 'title': 'z'},
  'aspectmode': 'manual',
  'aspectratio': {'x': 21.402828416162183, 'y': 4.894029008626602, 'z': 1.0}}}

aspectratio, height and width are all working but bgcolor doesn't work.

**Comment by jbednar** [comment]
I'm afraid I don't know anything about Plotly. Is 'bgcolor' a plot option available in Plotly?  I know bgcolor works in Bokeh, but I have no idea if that's what the option is called in Plotly, and if it's not, it shouldn't have any effect on Plotly apart from a warning message.

**Comment by GdMacmillan** [comment]
Unfortunately i see no warning messages. Plotly is the only backend i can use I think because my plots are dependent on the 3DScatter and 3DPath objects. It's OK. It works just fine, I just can't change the background color for some reason. Thanks for the help.

**Comment by jbednar** [comment]
You can do hv.help(hvobj) for any hvobj to see what options there are; Plotly probably calls it something other than `bgcolor`?

**Comment by philippjfr** [comment]
`bgcolor` is a plot option in holoviews, so it should be handled consistently across backends, but is probably not implemented.

**Comment by arabidopsis** [comment]
This might not be exactly the same problem but....
```
hv.output(backend='bokeh')
(hv.Box(0,0,10).opts(bgcolor='red')  * hv.Box(0,0,5).opts(bgcolor='blue')).opts(xlim=(-6,6),ylim=(-6,6))
```
Expecting: a blue box on a red box on a white plot.

Got: everything red, including the plot background.

This is the same with matplotlib backend too.

`hv.__version__ ==  '1.12.1'`

**Comment by philippjfr** [comment]
I can't reproduce any issue issue with 2D plots, the issue appears to be specific to plotly 3D plots.

@arabidopsis ``bgcolor`` controls the color of the background and the bottom layer takes precedence, if you want to color the boxes use color or fill_color.
